src/pkg/packager/validator/validator.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package validator contains functions for linting the zarf.yaml
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	zarfInvalidPrefix = "schema is invalid:"
+	zarfWarningPrefix = "zarf schema warning:"
+)
+
+// ValidateZarfSchema a zarf file against the zarf schema, returns an error if the file is invalid
+func ValidateZarfSchema(path string) (err error) {
+	var zarfTypedData types.ZarfPackage
+	if err := utils.ReadYaml(filepath.Join(path, layout.ZarfYAML), &zarfTypedData); err != nil {
+		return err
+	}
+
+	if err := checkForVarInComponentImport(zarfTypedData); err != nil {
+		message.Warn(err.Error())
+	}
+
+	zarfSchema, _ := config.GetSchemaFile()
+
+	var zarfData interface{}
+	if err := utils.ReadYaml(filepath.Join(path, layout.ZarfYAML), &zarfData); err != nil {
+		return err
+	}
+
+	if err = validateSchema(zarfData, zarfSchema); err != nil {
+		return err
+	}
+
+	message.Success("Validation successful")
```

Comment:
A more complete error message would be nice here - if you run `zarf prepare lint` with a zarf.yaml in the current directory you might not be sure which zarf.yaml it picked up (if you are less Zarf savvy)

Something like:
```suggestion
	message.Success("Schema validation successful for %q", zarfTypedData.Metadata.Name)
```

Reply:
returning err and messaging.Warn(err.Error()) could be more consistent :)

Reply:
Yeah agree that it's a bit weird that this function both returns error and uses the message package. Maybe I should have it not return an err and just do all the messaging within the function. Or should it return a custom type, with both warnings and errors. 

Curious what you guys think

Reply:
Yeah we're going to create a struct that has an array of warnings and an array of errors and return that instead.

Reply:
Ending up returning a validator object with warnings and errors and a regular error. The regular error is not null if something goes wrong in the validation process